rotkehlchen/assets/utils.py

```diff
@@ -247,17 +248,19 @@ def get_or_create_evm_token(
             else:
                 # inform frontend new token detected
                 data: dict[str, Any] = {'token_identifier': identifier}
-                if seen is not None:
+                if seen is not None and seen.should_notify:
                     if seen.tx_hash is not None:
                         data['seen_tx_hash'] = seen.tx_hash.hex()
                     else:  # description should have been given
                         data['seen_description'] = seen.description
                 if is_spam_token:
                     data['is_ignored'] = True
-                userdb.msg_aggregator.add_message(
-                    message_type=WSMessageType.NEW_EVM_TOKEN_DETECTED,
-                    data=data,
-                )
+
+                if seen is None or (seen and seen.should_notify is True):
+                    userdb.msg_aggregator.add_message(
+                        message_type=WSMessageType.NEW_EVM_TOKEN_DETECTED,
+                        data=data,
+                    )
```

Comment:
I think this diff can be made a tiny bit more readable. I tried to simplify even more but then I saw the edge case which made it complicated

```python
                # inform frontend new token detected
                data: dict[str, Any] = {'token_identifier': identifier}
                should_notify = False
                if is_spam_token:
                    data['is_ignored'] = True

                if seen is not None and seen.should_notify:
                    if seen.tx_hash is not None:
                        data['seen_tx_hash'] = seen.tx_hash.hex()
                    else:  # description should have been given
                        data['seen_description'] = seen.description
                    should_notify = True

                if seen is None or should_notify:
                    userdb.msg_aggregator.add_message(
                        message_type=WSMessageType.NEW_EVM_TOKEN_DETECTED,
                        data=data,
                    )
```